src/utilities/createRouterHistory.ts

```diff
@@ -42,7 +42,8 @@ export function createRouterHistory({ mode, listener }: RouterHistoryOptions = {
   let removeListener: (() => void) | undefined
 
   const startListening: () => void = () => {
-    if (!!listener && removeListener === undefined) {
+    if (listener) {
+      removeListener?.()
```

Comment:
IMO listener should be required and then we don't even need this if statement.

Reply:
listener comes from an options arg for history, I _really_ like options to be as optional as possible

Reply:
For user facing stuff for sure. But router literally doesn't work if you don't provide a listener here. It's not optional functionality wise